test/e2e/kubernetes/job/job.go

```diff
@@ -206,19 +206,29 @@ func AreAllJobsCompleted(jobPrefix, namespace string) (bool, error) {
 	return true, nil
 }
 
-// WaitOnReady is used when you dont have a handle on a job but want to wait until its in a Succeeded state.
-func WaitOnReady(jobPrefix, namespace string, sleep, duration time.Duration) (bool, error) {
+// WaitOnSucceeded returns true if all jobs matching a prefix substring are in a succeeded state within a period of time
+func WaitOnSucceeded(jobPrefix, namespace string, sleep, duration time.Duration) (bool, error) {
 	readyCh := make(chan bool, 1)
 	errCh := make(chan error)
```

Comment:
Could probably take these and turn them into one return struct with one channel.